src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs

```diff
@@ -170,6 +181,26 @@ public override bool Execute()
             return retVal;
         }
 
+        private string GetFullDirectoryPathWithSeperator(string directory)
+        {
+            string fullDirectoryPath = Path.GetFullPath(directory);
+
+            if (!fullDirectoryPath.EndsWith(Path.DirectorySeparatorChar.ToString()))
```

Comment:
`EndsWith(string)` should specify a comparison. A bit surprised at the `ToString()` allocation. Why not just look at the last character? 

```csharp
if (fullDirectoryPath.Length > 0 && fullDirectoryPath[^1] == Path.DirectorySeparatorChar)
```

Reply:
I was told to copy the code from here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs#L111 

It failed without the ToString(). Your option is probably better so I'll try that.

Reply:
@jaredpar this builds for netfx and your code doesn't compile for netfx.

Reply:
The joys of NetFX ... :( 

Should at least do the explicit string comparison part.